broker/src/main/java/org/apache/rocketmq/broker/transaction/TransactionalMessageCheckService.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.rocketmq.broker.transaction;
+
+import org.apache.rocketmq.broker.BrokerController;
+import org.apache.rocketmq.common.ServiceThread;
+import org.apache.rocketmq.common.constant.LoggerName;
+import org.apache.rocketmq.logging.InternalLogger;
+import org.apache.rocketmq.logging.InternalLoggerFactory;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class TransactionalMessageCheckService extends ServiceThread {
+    private static final InternalLogger log = InternalLoggerFactory.getLogger(LoggerName.TRANSACTION_LOGGER_NAME);
+
+    private BrokerController brokerController;
+
+    private final AtomicBoolean started = new AtomicBoolean(false);
+
+    public TransactionalMessageCheckService(BrokerController brokerController) {
+        this.brokerController = brokerController;
+    }
+
+    @Override
+    public void start() {
+        if (started.compareAndSet(false, true)) {
+            super.start();
+            this.brokerController.getTransactionalMessageService().open();
+        }
+    }
+
+    @Override
+    public void shutdown(boolean interrupt) {
+        if (started.compareAndSet(true, false)) {
+            super.shutdown(interrupt);
+            this.brokerController.getTransactionalMessageService().close();
+            this.brokerController.getTransactionalMessageCheckListener().shutDown();
+        }
+    }
+
+    @Override
+    public String getServiceName() {
+        return TransactionalMessageCheckService.class.getSimpleName();
+    }
+
+    @Override
+    public void run() {
```

Comment:
Should take advantage of what ServiceThread have already provided. No 
```
while(true)  {
...
sleep(xxx);
...
}
```

Use onWaitEnd() instead.

Reply:
Do you mean waitForRunning() and onWaitEnd() methods? yeah, I used these two methods to transform this implementation in the latest commit.